### PR TITLE
apis/tenancy: add validation to (Cluster)Workspace.spec.type

### PR DESCRIPTION
**Pull request by sttts**

### CODE REVIEW

File: pkg/apis/tenancy/v1alpha1/types.go [hunk 1]
```diff
@@ -84,6 +84,7 @@ type ClusterWorkspaceSpec struct {
 	//
 	// +optional
 	// +kubebuilder:default:="Universal"
+	// +kubebuilder:validation:Pattern=`^[A-Z][a-zA-Z0-9]*[a-zA-Z0-9]$`
```
Comment (ncdc) [inline pkg/apis/tenancy/v1alpha1/types.go:87]: Is this: at least 2 chars long, first must be capital letter, all subsequent are lowercase/uppercase/digit?

Comment (sttts) [reply pkg/apis/tenancy/v1alpha1/types.go:87]: CaML12Case321. Yes, 2 chars min length (we have that in a couple of places).

Comment (ncdc) [reply pkg/apis/tenancy/v1alpha1/types.go:87]: Can simplify regex. Use a single + instead of * then repeating the group

Comment (sttts) [reply pkg/apis/tenancy/v1alpha1/types.go]: updated

**Comment by qiujian16** [comment]
LGTM

**Review by qiujian16 — APPROVED** [review]

**Pull request merged**